testproject/Assets/Tests/Runtime/MultiprocessRuntime/Helpers/BuildMultiprocessTestPlayer.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.IO;
+#if UNITY_EDITOR
+using UnityEditor;
+using UnityEditor.Build.Reporting;
+#endif
+using UnityEngine;
+
+/// <summary>
+/// This is needed as Unity throws "An abnormal situation has occurred: the PlayerLoop internal function has been called recursively. Please contact Customer Support with a sample project so that we can reproduce the problem and troubleshoot it."
+/// when trying to build from Setup() steps in tests.
+/// </summary>
+public static class BuildMultiprocessTestPlayer
+{
+    public const string MultiprocessBaseMenuName = "MLAPI Multiprocess Test";
+    public const string BuildAndExecuteMenuName = MultiprocessBaseMenuName + "/Build - Execute multiprocess tests #%t";
+    public const string MainSceneName = "MultiprocessTestingScene";
+    public static string BuildPath => Path.Combine(Path.GetDirectoryName(Application.dataPath), "Builds/MultiprocessTestBuild");
+
+#if UNITY_EDITOR
+    [MenuItem(MultiprocessBaseMenuName+"/Build Test Player #t")]
+    public static void BuildNoDebug()
+    {
+        var success = Build();
+        if (!success)
+        {
+            throw new Exception("Build failed!");
+        }
+    }
+
+    [MenuItem(MultiprocessBaseMenuName+"/Build Test Player in debug mode")]
+    public static void BuildDebug()
+    {
+        var success = Build(true);
+        if (!success)
+        {
+            throw new Exception("Build failed!");
+        }
+    }
+
+    [MenuItem(MultiprocessBaseMenuName+"/Delete Test Build")]
+    public static void DeleteBuild()
+    {
+        switch (Application.platform)
+        {
+            case RuntimePlatform.WindowsPlayer:
+            case RuntimePlatform.WindowsEditor:
+                var exePath = $"{BuildPath}.exe";
+                if (File.Exists(exePath))
+                {
+                    File.Delete(exePath);
+                }
+                else
+                {
+                    Debug.Log($"exe {exePath} doesn't exist");
+                }
+                break;
```

Comment:
on Windows, build also creates some other artifacts such as `testproject_Data` folder, `UnityPlayer.dll`, `UnityCrashHandler64.exe` etc.
I'm not sure whether or not you tested this flow on Windows but I think we should delete the root `Builds/Tests/` folder entirely to have a full cleanup — and that wouldn't even need to be platform dependant since we'd be deleting everyting underneath the folder.
what do you think about this idea?

Reply:
my goal isn't to cleanup here, it's really to prevent execution of old builds because there was a failure previously.
I would expect workspace cleanup to be done at the yamato/bokken level afterward for all files, not just builds